lib/superform/rails/components/collection_select.rb

```diff
@@ -0,0 +1,84 @@
+module Superform
+  module Rails
+    module Components
+
+      # Replacement for :
+      #
+      #   <%= form.collection_select :point_id, Point.all, :id, :name, prompt: 'Select something'
+      #
+      # Usage - Based on https://apidock.com/rails/ActionView/Helpers/FormTagHelper/select_tag
+      #
+      #   component = field(:person_id).select(collection: @people, value: :id, text: :name, selected: selected: model.person_id)
+      #
+      # For efficiency can also use pluck as follows
+      #
+      #   component = field(:jurisdiction_id).select(collection: Jurisdiction.all.pluck(:id, :name), value: :first, text: :last)
+      #
+      # Can also use shorter form (without specifying text and value methods) :
+      #
+      #   select(collection:collection: Jurisdiction.all.pluck(:name, :id).map { |j| [j[0], j[1]] }, options: { class: 'app-input' })
+      #
+      # Options
+      #
+      #   component = field(:person_id).select(collection: @people, value: :id, text: :name,
+      #                                        options: { allow_blank: true, prompt: 'Select a Person' } )
+      #
+      # Example in a Form
+      #
+      #   div class: "w-[300px] field flex-1 mb-2 mt-6" do
+      #     component = field(:jurisdiction_id).select(collection: Jurisdiction.all.pluck(:name, :id).map { |j| [j[0], j[1]] },
+      #                                                selected: model.jurisdiction_id,
+      #                                                options: { class: 'app-input' })
+      #
+      #     render component.field.label(class: "font-text-regular-base text-color-brand-ui-300 mb-2")
+      #     render component
+      #   end
+      #
+      class CollectionSelect < FieldComponent
+
+        # The array of model instances to display as set of <option> tags
+        # If no text or value methods are provided then the collection is expected to already be an Array of Array,
+        # where each entry contains [TEXT, VALUE] e.g [[name:, :id], [name:, :id] ... ]
+        #
+        def collection
+          @attributes.fetch(:collection, [])
+        end
+
+        def options
+          @attributes.fetch(:options, {})
+        end
+
+        # Which of the options, if any, have already been selected
+        def selected
+          @attributes.fetch(:selected, nil)
+        end
+
+        # The method to call on collection instances to get text to display in the option Dropdown
+        def text
+          @attributes.fetch(:text, nil)
+        end
+
+        # The method to call on collection instances to generate the option value
+        # Defaults to the Id
+        #
+        def value
+          @attributes.fetch(:value, :id)
+        end
+
+        def template(&)
```

Comment:
This library should avoid calling into Rails form helpers (so far I've been successful at that) because they're not really composable.

Here's what the public API would look like for `select`:

```ruby
render field(:collection_point).select { CollectionPoint.all { |cp| option(value: cp.id) { cp.name } }
# Or it might look like this; I can't recall if Phlex needs an argument in the block.
render field(:collection_point).select { |s| CollectionPoint.all { |cp| s.option(value: cp.id) { cp.name } }
```

This will maintain the composability that Phlex offers. Another interesting thing about this is that the `SelectComponent` class will have no knowledge of the collection—instead the collection is populated within the `select {}` block via the `option` tag.

Another goal of Superform is to "stay as close to the HTML as possible", which means non-HTML arguments like `select(collection: [])` should be avoided for the base API since `collection` isn't a valued attribute for the `select` tag.

Reply:
These changes will make the `SelectComponent` class much smaller than it currently is implemented.

Reply:
Also, if the API above seems verbose for collection, it is. That's needed though to make the following "shortcut" possible:

```ruby
render field(:collection_point).select.options(CollectionPoint, &:name)
```

Reply:
Ok, I'll try to simplify the collection aspect, but how should the options such as  - selected option, the prompt, multiple treu/false and html options such as a class be handled ?

Reply:
All the HTML attributes & tags get handled via the Phlex components:

```ruby
render field(:collection_point).select(class: "p4 bg-red-100") { |s| CollectionPoint.all { |cp| s.option(value: cp.id) { cp.name } }
```

Selected is awkward (for now), but would look like this:

```ruby
render field(:collection_point).select do |s|
  CollectionPoint.all do |cp| 
    s.option(value: cp.id, selected: (cp.id == field(:collection_point).value)) { cp.name }
  end
end
```

A true/false select would look like:

```ruby
render field(:collection_point).select do |s|
  option(value: "true") { "True" }
  option(value: "false") { "False" }
end
```

That could be put behind something like `render field(:is_admin).select_boolean` if it's common to have a drop/down for t/f values. Does Rails do that already? I've never tried it.

By "the prompt" do you mean the label?